IdentityCore/src/network/MSIDURLSessionManager.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
+@class MSIDURLSessionDelegate;
+
+@interface MSIDURLSessionManager : NSObject
+
+- (instancetype _Nullable)initWithConfiguration:(nonnull NSURLSessionConfiguration *)configuration
+                                        delegate:(nullable MSIDURLSessionDelegate *)delegate NS_DESIGNATED_INITIALIZER;
+
+@property (class, nonnull) MSIDURLSessionManager *defaultManager;
+@property (nonatomic, readonly, nonnull) NSURLSessionConfiguration *configuration;
+@property (nonatomic, readonly, nonnull) NSURLSession *session;
+@property (nonatomic, readonly, nullable) MSIDURLSessionDelegate *delegate;
```

Comment:
nit: Why do we need to save delegate outside of passing it to NSURLSession?

Reply:
you are right, removed it.